src/main/java/org/mule/extension/ftp/internal/ftp/command/ClassicFtpCommand.java

```diff
@@ -176,4 +180,43 @@ public RuntimeException exception(String message, Exception cause) {
   private String enrichExceptionMessage(String message) {
     return format("%s. Ftp reply code: %d", message, client.getReplyCode());
   }
+
+  private Optional<FTPFile> getFileFromPath(Path path) throws IOException {
+    String filePath = path.toAbsolutePath().toString();
+    // Check if MLST command is supported
+    FTPFile file = client.mlistFile(filePath);
+    if (file == null) {
+      FTPFile[] files = client.listFiles(filePath);
+      if (files.length >= 1) {
+        if (filePath.endsWith(files[0].getName())) {
+          // List command result is the file from the path parameter
+          file = files[0];
+        } else {
+          // List command result is a directory
+          if (path.getParent() != null) {
+            files = client.listDirectories(path.getParent().toAbsolutePath().toString());
+            return Arrays.stream(files).filter(dir -> filePath.endsWith(dir.getName())).findFirst();
+          } else {
+            // root directory
+            file = createRootFile();
```

Comment:
I know (not glad about this), but in those servers in which MLST is not supported, we don't have (or at least I could find) any other means to obtain root's metadata. And because we are using this mechanism internally to tell if a file/directory exists or not (and the root always exists), I found this unfaithful idea to be the one that worked faster (with less amount of changes) for most use cases.

Anyways, if the user access root directory attributes... he will find out about this charade :'(